boa_parser/src/lexer/identifier.rs

```diff
@@ -93,11 +93,20 @@ impl<R> Tokenizer<R> for Identifier {
         let (identifier_name, contains_escaped_chars) =
             Self::take_identifier_name(cursor, start_pos, self.init)?;
 
-        let token_kind = match identifier_name.parse() {
+        let parsed_token = identifier_name.parse();
+
+        if parsed_token.is_ok() && contains_escaped_chars {
+            return Err(Error::Syntax(
+                "Keyword must not contain escaped characters.".into(),
+                start_pos,
+            ));
+        }
+
+        let token_kind = match parsed_token {
             Ok(Keyword::True) => TokenKind::BooleanLiteral(true),
             Ok(Keyword::False) => TokenKind::BooleanLiteral(false),
             Ok(Keyword::Null) => TokenKind::NullLiteral,
```

Comment:
We shouldn't disallow escaped keywords in all circumstances only for expression parsing of keywords.

which happens here here, so we disallow it here.

Reply:
Oh, I thought all keywords cannot contain any unicode escape based on NOTE 1 from the [spec](https://tc39.es/ecma262/#sec-keywords-and-reserved-words). Is my interpretation incorrect?

Reply:
> Oh, I thought all keywords cannot contain any unicode escape based on NOTE 1 from the [spec](https://tc39.es/ecma262/#sec-keywords-and-reserved-words). Is my interpretation incorrect?

I meant our [`Keyword`](https://github.com/boa-dev/boa/blob/main/boa_ast/src/keyword/mod.rs) enum type (which doesn't conform to the spec), contains `Let`, `private`, etc. which can be identifiers and can have escape sequences.

So disallowing all of them breaks valid code. That's I suggested only disallowing some. the ones that are part of the `ReservedWords` (expect `async` and `await` as the spec says) can be identifiers.